classes/file_manager.php

```diff
@@ -0,0 +1,186 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Class to manage seb files, used in part to abstract some of the Moodle file manager logic.
+ *
+ * @package    quizaccess_seb
+ * @author     Andrew Madden <andrewmadden@catalyst-au.net>
+ * @copyright  2019 Catalyst IT
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace quizaccess_seb;
+
+use stored_file;
+
+defined('MOODLE_INTERNAL') || die();
+
+class file_manager {
+
+    /** Component the file will be stored in.  */
+    const COMPONENT = 'mod_quiz';
```

Comment:
is it correct?

Reply:
Hadn't fully registered what the component and filearea vars were for. Was only guessing from the code. Just clicked that they are the plugin (component) and part of plugin (filearea) indicators. Updated to `quizaccess_seb`. 

Should filearea be changed to `quizsettings` from `quizaccess_seb_quizsettings`? `quizaccess_seb_quizsettings` is the name of the table they are stored in.